### PR TITLE
feature/6-FontendCrud

### PR DESCRIPTION
**Pull request by yaroslav-begar**

### CODE REVIEW

**Comment by yaroslav-begar** [comment]
[Lesson6-4](https://trello.com/c/6199yFkj/54-lesson6-4)

File: app/code/Test/CustomerVouchers/Controller/Voucher/Save.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    /**
+     * @return \Magento\Framework\App\ResponseInterface
+     * @throws \Magento\Framework\Exception\AlreadyExistsException
+     * @throws \Magento\Framework\Exception\LocalizedException
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    public function execute()
+    {
+        if ($this->getRequest()->isPost()) {
```
Comment (epolish) [inline app/code/Test/CustomerVouchers/Controller/Voucher/Save.php:47]: You should also use CSRF validation.
https://magento.stackexchange.com/questions/187061/magento-2-form-key/187083

File: app/code/Test/CustomerVouchers/view/frontend/templates/Voucher/insert.phtml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+<h3>Add voucher</h3>
+
+<br /><br />
+
+<?php $record = $block->getRecord(); ?>
+
+<form action="<?= $block->getFormAction(); ?>" method="post">
```
Comment (epolish) [inline app/code/Test/CustomerVouchers/view/frontend/templates/Voucher/insert.phtml:7]: http://mageprince.com/blog/how-to-add-form-key-in-magento-2/

File: app/code/Test/CustomerVouchers/Observer/ActionObserver.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    /**
+     * @param \Magento\Framework\Event\Observer $observer
+     */
+    public function execute(\Magento\Framework\Event\Observer $observer)
+    {
+        $controllerName  = $observer->getControllerAction()->getRequest()->getControllerName();
+
+        if (in_array($controllerName, ['voucher', 'voucher_status']) && !$this->customerSession->isLoggedIn()) {
```
Comment (epolish) [inline app/code/Test/CustomerVouchers/Observer/ActionObserver.php:33]: This logic should be inside controllers. You can check there, if customer session doesn't persist, redirect (or forward) to login action.

**Review by epolish — APPROVED** [review]

**Pull request merged**